sdk/containerregistry/container-registry/src/content/models.ts

```diff
@@ -98,16 +98,30 @@ export interface OciDescriptor {
 }
 
 /**
- * Helper used to determine whether a given manifest downloaded using ContainerRegistryBlobClient.getManifest() is an OCI image manifest.
+ * Helper used to determine whether a given manifest downloaded using ContainerRegistryBlobClient.getManifest() satisfies the OCI image manifest schema.
  */
 export function isOciImageManifest(
   manifest: Record<string, unknown>
 ): manifest is OciImageManifest {
-  return (
-    manifest.schemaVersion === 2 &&
-    typeof manifest.config === "object" &&
-    Array.isArray(manifest.layers)
-  );
+  // schemaVersion must be 2
+  if (manifest.schemaVersion !== 2) {
+    return false;
+  }
+
+  // Media type can be unset, but if it is set it must match the OCI image manifest media type
+  if (manifest.mediaType && manifest.mediaType !== KnownManifestMediaType.OciImageManifest) {
```

Comment:
[application/vnd.docker.distribution.manifest.v2+json](https://github.com/distribution/distribution/blob/master/docs/spec/manifest-v2-2.md#image-manifest-field-descriptions)](https://github.com/opencontainers/image-spec/blob/main/media-types.md#applicationvndociimagemanifestv1json) can be represented with `OciImageManifest` and is fully compatible, should we add it here and maybe rename to `isImageManifest`?

Reply:
Discussed with Liudmila offline. We're going to remove the type guard entirely in favor of users checking the `mediaType` property on `GetManifestResult` and then casting `manifest` to `OciImageManifest` themselves. This is more in line with the approach other languages are taking, and makes a bit more sense, since this type guard is not enough to ensure that a manifest is an OCI image manifest.